test/sql/function/list/list_concat.test

```diff
@@ -187,6 +187,22 @@ SELECT array_push_front(NULL, 1);
 ----
 [1]
 
+# concat operator
+query T
+SELECT [1, 2] || NULL
```

Comment:
Can we add a check that the type of this is int32[]? In the original issue I filed, this is currently getting typed as int32

Reply:
It's getting correctly typed as `NULL`:

```sql
D select typeof([1,2] || NULL) AS type;
┌─────────┐
│  type   │
│ varchar │
├─────────┤
│ "NULL"  │
└─────────┘
```

However, DuckDB doesn't currently return null types since not all export formats support them - and instead converts NULL to an `int32` when providing the final result.